internal/internal_update.go

```diff
@@ -130,7 +130,7 @@ func (up *updateProtocol) requireState(action string, valid ...updateState) {
 			return
 		}
 	}
-	panicIllegalState(fmt.Sprintf("invalid action %q in update protocol %v", action, up))
+	panicIllegalState(fmt.Sprintf("invalid action %q in update protocol %+v", action, up))
```

Comment:
Does this occur for user mistakes or just mistakes we've made internal to the SDK/server? If the former, I would prefer more structure to the message

Reply:
Mistakes we have made